internal/api/user.go

```diff
@@ -129,8 +126,13 @@ func (a *API) UserUpdate(w http.ResponseWriter, r *http.Request) error {
 		var terr error
 		if params.Password != nil {
 			if config.Security.UpdatePasswordRequireReauthentication {
-				if terr = a.verifyReauthentication(params.Nonce, tx, config, user); terr != nil {
-					return terr
+				now := time.Now()
+
+				// we require reauthentication if the user hasn't signed in recently in the current session
+				if session == nil || now.After(session.CreatedAt.Add(24*time.Hour)) {
```

Comment:
Hmm i'm hesitant to make this a config because i don't want to overcrowd the dashboard - was thinking we can go with this first until we get user feedback that they don't want this.